src/dex/uniswap-v2/ring-v2.ts

```diff
@@ -0,0 +1,349 @@
+import { defaultAbiCoder, Interface } from '@ethersproject/abi';
+import { getAddress } from '@ethersproject/address';
+import { keccak256 } from '@ethersproject/keccak256';
+import {
+  DEST_TOKEN_PARASWAP_TRANSFERS,
+  Network,
+  SRC_TOKEN_PARASWAP_TRANSFERS,
+  SwapSide,
+} from '../../constants';
+import { IDexHelper } from '../../dex-helper';
+import { SpecialDex } from '../../executor/types';
+import { applyTransferFee } from '../../lib/token-transfer-fee';
+import {
+  Token,
+  TransferFeeParams,
+  ExchangePrices,
+  Address,
+  DexExchangeParam,
+  NumberAsString,
+  DexConfigMap,
+} from '../../types';
+import { getBigIntPow, getDexKeysWithNetwork } from '../../utils';
+import { DexParams, UniswapData, UniswapV2Data } from './types';
+import { UniswapV2 } from './uniswap-v2';
+import ringV2ABI from '../../abi/ring-v2/ring-v2-pool.json';
+import ringV2factoryABI from '../../abi/ring-v2/ring-v2-factory.json';
+import RingV2ExchangeRouterABI from '../../abi/ring-v2/ring-v2-router.json';
+import ETHMainnetFewWrappedTokenJSON from '../../abi/ring-v2/few-wrapped-token.json';
+import { extractReturnAmountPosition } from '../../executor/utils';
+
+export enum RingV2Functions {
+  swapExactTokensForTokens = 'swapExactTokensForTokens',
+  swapExactETHForTokens = 'swapExactETHForTokens',
+  swapTokensForExactTokens = 'swapTokensForExactTokens',
+  swapTokensForExactETH = 'swapTokensForExactETH',
+  swapETHForExactTokens = 'swapETHForExactTokens',
+}
+
+const RingV2Config: DexConfigMap<DexParams> = {
+  RingV2: {
+    [Network.MAINNET]: {
+      factoryAddress: '0xeb2A625B704d73e82946D8d026E1F588Eed06416',
+      initCode:
+        '0xa7ae6a5ec37f0c21bbdac560794258c4089b8ae3ffa6e3909b53c6091764a676',
+      poolGasCost: 80 * 1000,
+      feeCode: 30,
+      router: '0x39d1d8fcC5E6EEAf567Bce4e29B94fec956D3519',
+    },
+  },
+};
+
+const FewWrappedTokenConfig: DexConfigMap<{
+  fewWrapFactory: string;
+  bytecode: string;
+}> = {
+  RingV2: {
+    [Network.MAINNET]: {
+      fewWrapFactory: '0x7D86394139bf1122E82FDF45Bb4e3b038A4464DD',
+      bytecode: ETHMainnetFewWrappedTokenJSON.bytecode,
+    },
```

Comment:
Can we simply hardcode `initCode` instead of storing `bytecode` and calculating it each time?